### DISCUSSION TITLE
How to pass logged in user's email to Google Analytics?

### DISCUSSION THREAD

**Discussion by itrixinc** [filamentphp/filament#5199] [Help]
I am a newbie to filamentphp.

Filament::registerScripts([
            asset('js/gtag.js'),
        ]);

I have registered the js that loads the google analytics js. 
But in that JS how can I pass the current auth()->user()-email?

using: chiiya/filament-access-control [spatie/laravel-permission]

--- I am attaching the gtag.js, in case this helps someone ----
gtag.js
--------------------------------------------------------------------------------------------------------------
var host = window.location.hostname;
if(host == "live.hostname")
{
    var imported = document.createElement('script');
	imported.src = 'https://www.googletagmanager.com/gtag/js?id=G-XXXXXX-ID';
	document.head.appendChild(imported);

	window.dataLayer = window.dataLayer || [];
	function gtag(){dataLayer.push(arguments);}
	gtag('js', new Date());

	gtag('config', 'G-XXXXXX-ID', {
        'user_id': 'USER_ID'    // <--- This is where I would like to pass the current logged in user-email [auth()->user()-email]
        });
}

**Comment by itrixinc** [comment]
I am still stuck on this...Any pointers ?

**Comment by ArtMin96** [comment]
Possible options that you can try.

1) With render hook

[https://filamentphp.com/docs/2.x/admin/appearance#render-hooks](https://filamentphp.com/docs/2.x/admin/appearance#render-hooks)

`scripts.start` - before scripts are defined
`scripts.end` - after scripts are defined

Add blade view that contains scripts

2) `Filament::registerScripts()`
